### ISSUE TITLE
Places API (new): FieldMask is required required parameter but really unclear how to supply it

### ISSUE THREAD

**Issue by theRocket** [googleapis/google-cloud-python#12083]
Calling the sample code at `samples/generated_samples/places_v1_generated_places_get_place_sync.py` with the API key supplied and a valid Place ID, such as:
```
    client = places_v1.PlacesClient(
        client_options=options,
    )
    # Initialize request argument(s)
    request = places_v1.GetPlaceRequest(
        name="ChIJWyLAeY-jhVQRJXqchdJTMio",
    )
    # Make the request
    response = client.get_place(request=request)
```
returns the following exception:
```
File "google-cloud-python/packages/google-maps-places/samples/generated_samples/places_v1_generated_places_get_place_sync.py", line 55, in <module>
    sample_get_place(MY_API_KEY)
  File "google-cloud-python/packages/google-maps-places/samples/generated_samples/places_v1_generated_places_get_place_sync.py", line 49, in sample_get_place
    response = client.get_place(request=request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.pyvenv/gcp-env/lib/python3.12/site-packages/google/maps/places_v1/services/places/client.py", line 846, in get_place
    response = rpc(
               ^^^^
  File "~/.pyvenv/gcp-env/lib/python3.12/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
    return wrapped_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.pyvenv/gcp-env/lib/python3.12/site-packages/google/api_core/grpc_helpers.py", line 77, in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.InvalidArgument: 400 FieldMask is a required parameter. See https://cloud.google.com/apis/docs/system-parameters on how to provide it. As an example, you can set the header 'X-Goog-FieldMask' to value 'places.displayName', 'places.id' to ask for the display name and the place id of a place. You can also set the value to '*' in manual testing to get all the available response fields.
```
yet that documentation says:

> Most users will not need to use these parameters directly. However, they are often used by Google-provided client libraries.

So where does this value go, and in what format?

More docs on valid field types:
https://developers.google.com/maps/documentation/places/web-service/choose-fields

> You specify the field list by creating a response field mask. You then pass the response field mask to either method by using the parameter $fields or fields, or by using the HTTP or gRPC header X-Goog-FieldMask.

**Comment by parthea** [comment]
Hi @theRocket,

The `x-goog-fieldmask` request header can be set via the `metadata` argument of `PlacesClient.get_place()` which is used for adding custom request headers. For example, 

```
    # Make the request
    response = client.get_place(request=request, metadata[("x-goog-fieldmask", "id,displayName")])
```

The format of the value for `x-goog-fieldmask` header should be the same as shown in the REST docs [here](https://developers.google.com/maps/documentation/places/web-service/place-details#GetPlaceRequest), for example when using `curl`.

```
curl -X GET -H 'Content-Type: application/json' \
-H "X-Goog-Api-Key: API_KEY" \
-H "X-Goog-FieldMask: id,displayName" \
https://places.googleapis.com/v1/places/ChIJWyLAeY-jhVQRJXqchdJTMio
```

https://github.com/googleapis/google-cloud-python/blob/f4938c05ef84a93bf05b6012053baece659caa63/packages/google-maps-places/google/maps/places_v1/services/places/client.py#L103-L108

https://github.com/googleapis/google-cloud-python/blob/f4938c05ef84a93bf05b6012053baece659caa63/packages/google-maps-places/google/maps/places_v1/services/places/client.py#L753-L760

https://github.com/googleapis/google-cloud-python/blob/f4938c05ef84a93bf05b6012053baece659caa63/packages/google-maps-places/google/maps/places_v1/services/places/client.py#L805-L806

**Comment by theRocket** [comment]
Thank you, @parthea! I was getting close to that solution, but the guess & check was frustrating me. Here is the final code snippet that works:
```
    # Initialize request argument(s)
    request = places_v1.GetPlaceRequest(
        name="places/ChIJWyLAeY-jhVQRJXqchdJTMio",
    )
    fieldMask = "formattedAddress,displayName"

    # Make the request
    response = client.get_place(request=request, metadata=[("x-goog-fieldmask",fieldMask)])
```
I

**Comment by theRocket** [comment]
Another issue with `SearchNearbyRequest` being very obtuse about proper inputs. Can you help, @parthea? Several attempts at this in various data type constructs have failed:

```
home_loc = [48.7588228,-122.4634339]

def sample_search_nearby(api_key, lat_lng = home_loc, rad=10.0):
    # Create a client
    options = ClientOptions(api_key=api_key)
    client = places_v1.PlacesClient(client_options=options)

    # Initialize request argument(s)
    search_km = 1000*rad # convert from km to meters, max 50000
    search_circle = places_v1.types.Circle(center=lat_lng, radius=search_km)
    loc_restriction = places_v1.SearchNearbyRequest.LocationRestriction(search_circle)

    request = places_v1.SearchNearbyRequest(
        location_restriction=loc_restriction,
        included_types=['electric_vehicle_charging_station']
    )

    # Make the request
    response = client.search_nearby(request=request)
```
Error response:
```
  File "/Users/ricker/Code/GCP/google-cloud-python/packages/google-maps-places/samples/generated_samples/places_v1_generated_places_search_nearby_sync.py", line 50, in sample_search_nearby
    search_circle = places_v1.types.Circle(center=lat_lng, radius=search_km)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ricker/.pyvenv/gcp-env/lib/python3.12/site-packages/proto/message.py", line 604, in __init__
    super().__setattr__("_pb", self._meta.pb(**params))
                               ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Message must be initialized with a dict: google.maps.places.v1.Circle
```
This is a required field and the `google.maps.places_v1.types.geometry.Circle` type signature suggests I have initialized it correctly. I tried also passing a dictionary but received the same error.

If I try:
```
search_circle = Circle(lat_lng, search_km)
```
I get this error:
```
TypeError: Message.__init__() takes from 1 to 2 positional arguments but 3 were given
```
I may be new to python but it looks like I passed 2 arguments.
```

**Comment by theRocket** [comment]
Sorry, I took one more stab and found the right construct. It is a dictionary for `LatLng`:
```python
home_loc = {'latitude':48.7649179, 'longitude':-122.4602791}

def sample_search_nearby(api_key, lat_lng = home_loc, rad=10.0):
    ...
    # Initialize request argument(s)
    search_km = 1000*rad # convert from km to meters, max 50000
    loc_restriction = places_v1.SearchNearbyRequest.LocationRestriction()
    loc_restriction.circle = places_v1.Circle(center=home_loc, radius=search_km)
```

**Comment by talksik** [comment]
Same problem in https://github.com/googleapis/google-cloud-go

Really confusing telling someone to adjust headers somehow when the SDK initializes a RestClient for you.

**Comment by theRocket** [comment]
> Same problem in https://github.com/googleapis/google-cloud-go

I suspect because these are both generated clients from the same [Protobuf implementation](https://protobuf.dev/overview/).

Regarding programming languages, Google appears to attempting to solve the primary problem of writing language-agnostic code only once so consumers can use whatever flavor of the month from the following list:

- [C++](https://protobuf.dev/reference/cpp/cpp-generated#invocation)
- [C#](https://protobuf.dev/reference/csharp/csharp-generated#invocation)
- [Java](https://protobuf.dev/reference/java/java-generated#invocation)
- [Kotlin](https://protobuf.dev/reference/kotlin/kotlin-generated#invocation)
- [Objective-C](https://protobuf.dev/reference/objective-c/objective-c-generated#invocation)
- [PHP](https://protobuf.dev/reference/php/php-generated#invocation)
- [Python](https://protobuf.dev/reference/python/python-generated#invocation)
- [Ruby](https://protobuf.dev/reference/ruby/ruby-generated#invocation)
- [Dart](https://github.com/google/protobuf.dart)
- [Go](https://github.com/protocolbuffers/protobuf-go)

However, making the use of these clients in a well-documented, human understandable way (i.e. developer friendly) seems to be a secondary concern. Perhaps they expect AI to be writing all the code to consume their APIs in the near future.
